lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,323 @@
+
+    def _prepare_common_vhd(self, vhd_local_path: PurePath, log: Logger) -> None:
+        if self._common_vhd_path:
+            return
+
+        vhd_remote_path = PureWindowsPath(
+            self.server_node.working_path / f"common_vhd.{vhd_local_path.suffix}"
+        )
+
+        is_zipped = False
+        if vhd_local_path.suffix == ".zip":
+            is_zipped = True
+            vhd_remote_path = PureWindowsPath(
+                self.server_node.working_path / "zipped_vhd.zip"
+            )
+
+        log.debug("Copying VHD to server")
+        self.server_node.shell.copy(vhd_local_path, vhd_remote_path)
+        log.debug("Finished copying VHD to server")
+
+        if is_zipped:
+            vhd_remote_path = self._unzip_vhd(vhd_remote_path)
+
+        self._common_vhd_path = vhd_remote_path
+
+    def _deploy_nodes(self, environment: Environment, log: Logger) -> None:
+        if environment.runbook.nodes_requirement is None:
+            return  # nothing to deploy?
+
+        test_suffix = "".join(random.choice(string.ascii_uppercase) for _ in range(5))
+        vm_name_prefix = f"lisa-{test_suffix}"
+
+        hv = self.server_node.tools[HyperV]
+        default_switch = hv.get_first_switch()
+
+        extra_args = {
+            x.command.lower(): x.args for x in self._hyperv_runbook.extra_args
+        }
+
+        self.console_logger = SerialConsoleLogger(self.server_node)
+
+        for i, node_space in enumerate(environment.runbook.nodes_requirement):
+            node_runbook = node_space.get_extended_runbook(
+                HypervNodeSchema, type(self).type_name()
+            )
+
+            vm_name = f"{vm_name_prefix}-{i}"
+
+            node = environment.create_node_from_requirement(node_space)
+            assert isinstance(node, RemoteNode)
+
+            self._prepare_common_vhd(PurePath(node_runbook.vhd), log)
+            assert self._common_vhd_path
+
+            node.name = vm_name
+
+            node_context = get_node_context(node)
+            node_context.vm_name = vm_name
+            node_context.host = self.server_node
+
+            node_context.working_dir = PureWindowsPath(
+                self.server_node.working_path / f"{vm_name}"
+            )
+
+            vm_vhd_name = f"{vm_name}.{self._common_vhd_path.suffix}"
+            node_context.vhd_path = PureWindowsPath(
+                self.server_node.working_path / f"{vm_vhd_name}"
+            )
+
+            self.server_node.tools[Mkdir].create_directory(
+                str(node_context.working_dir)
+            )
+
+            self.server_node.tools[Cp].copy(
+                self._common_vhd_path, node_context.vhd_path
+            )
+
+            self._resize_vhd_if_needed(node_context.vhd_path, node_runbook)
+
+            assert isinstance(node.capability.core_count, int)
+            assert isinstance(node.capability.memory_mb, int)
+
+            com1_pipe_name = f"{vm_name}-com1"
```

Comment:
The name could be ```com1_serial_log = f"{vm_name}-serial-log"```

Reply:
This is the name of the pipe corresponding to the COM port. IMO it doesn't make sense to add `-serial-log` to it. The name `{vm_name}-com1` says that this pipe corresponds to the COM1 port of the VM with name `vm_name`.